src/Factory.php

```diff
@@ -418,144 +434,6 @@ function(self $factory): self {
         );
     }
 
-    private function ownedRelationshipField(?string $relationName, self $factory): ?string
```

Comment:
all of these methods were just actually digging into doctrine's metadata... I've replaced them with `getRelationshipMetadata()`